samples/net/sockets/socketpair/src/socketpair_example.cpp

```diff
@@ -0,0 +1,184 @@
+/*
```

Comment:
Why C++? At least I would prefer a plain C program as a sample.

Reply:
> Why C++? At least I would prefer a plain C program as a sample.

I'm kinda surprised too. But without people bringing up new ideas, we won't develop as wide and far as we could. We could treat this sample as a cross between showcases socketpair and C++ functionality, and we clearly need more C++ samples. And if this sample really builds under both a POSIX system and Zephyr, IMHO that's rather positive development overall.

Reply:
I find C++ just a nice shorthand for C requiring fewer lines of code - especially when looking at the various container types. It's more expressive.
I could potentially convert it to C, but at the same time.. if it Just Works™.
It's just a sample program but I too thought it would be good to start demonstrating C++ code running on Zephyr too from a code diversity perspective. There are a lot of devs who *want* to write C++ code on their microcontrollers, so maybe it would be good advertisement ¯\_(ツ)_/¯.

Reply:
I would like to see a sample implemented using C like the other network samples. That way anyone writing a C program can use this sample right away. Having a C++ sample here is fine too, we can have two implementations.

Reply:
> we can have two implementations.

@cfriedt, So I hope you'll keep C++ variant, as I mentioned, IMHO it's useful.

Reply:
I have a separate PR now for multithreading in C++, so adding some samples for C++ application support is a no-brainer. I'll rewrite this one in C.